### DISCUSSION TITLE
After modifying a column in clikchouse, does the connector need to be restarted?

### DISCUSSION THREAD

**Discussion by zTranquilizer** [ClickHouse/clickhouse-kafka-connect#475] [Q&A] [answered]
Hi all, noticed this peculiarity and can't figure out if it's something I have that's not working right or if it's the connector that's working that way.

If I have modified the column data type
```
ALTER TABLE TestTable MODIFY COLUMN IF EXISTS DateCreated DateTime
```
 do I also need to restart the connector/task ?
 
When I restart - everything starts working fine. But before the restart the connector doesn't realize that the data schema has changed, although I can see the log: “Update table mapping” or something like that. 
Also in the connector there is a setting `tableRefreshInterval`, I thought it is responsible for pulling up the new schema in real time, and does not require restarting the connector, but maybe I'm wrong.

Also another point, almost unrelated to the question. I saw that the `DateTime` data type is supported by the connector, but when I add a small change, something like `DateTime('UTC')` and it stops working. I take it `DateTime('UTC')` is not supported by the connector yet?

**Comment by Paultagoras** [comment, marked as answer]
In general, it's safer to stop the connector -> alter the table -> start the connector.

tableRefreshInterval sets the time for refreshing the table mapping, but there is usually a gap between updates.

For DateTime, what is your data format?

**Reply by zTranquilizer** [reply]
> In general, it's safer to stop the connector -> alter the table -> start the connector.

Thank you.

>tableRefreshInterval sets the time for refreshing the table mapping, but there is usually a gap between updates.

Ok, I thought it somehow pulls up the changes to the database schema and then works properly in connection with this information, maybe I don't fully understand the concept. I thought thanks to this setting after updating the database schema it is not necessary to reload the connector at all.

> For DateTime, what is your data format?

Protobuf, if that's what you mean.
